File: Python/92. Reverse Linked List II.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseBetween(self, head, left, right):
        """
        :type head: ListNode
        :type left: int
        :type right: int
        :rtype: ListNode
        """
        dummy = ListNode(0)
        dummy.next = head
        prev = dummy
        for i in range(left - 1): 
            prev = prev.next 
        cur, next = prev.next, prev.next.next
        for i in range(right - left):
            cur.next = next.next
            next.next = prev.next
            prev.next = next
            next = cur.next
        return dummy.next
        
        

File: C#/92. Reverse Linked List II.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseBetween(ListNode head, int left, int right) {
        var dummy = new ListNode(-1);
        dummy.next = head;
        
        ListNode prev = dummy;
        for(var i = 0; i < left - 1; i++) {
            prev = prev.next;
        }
        
        ListNode cur = prev.next;
        for (var i = left; i < right; i++) {
            var next = cur.next;
            cur.next = next.next;
            next.next = prev.next;
            prev.next = next;
        }
        
        return dummy.next;
    }
}